### ISSUE TITLE
[question] mocha and path mappings

### ISSUE THREAD

**Issue by fredrik-w** [aspect-build/rules_ts#227]
I'm trying to migrate our mono-repo from rules_nodejs to rules_js/rules_ts and have made good progress. However, running tests with mocha doesn't work and I believe they are executed in a different manner than how it is done in rules_nodejs.

We're using paths in tsconfig to do cross-module imports which works when building. But when running the tests they fail and claim the files aren't found. Looking at the output-directory I can see that all needed files are transpiled and part of the runfiles-set. So I'm thinking it must be something with _how_ the tests are executed.

In our `WORKSPACE` file I've added the following for the `bins` option in `npm_translate_lock`:
```starlark
        # derived from "bin" attribute in node_modules/mocha/package.json
        "mocha": {
            "mocha": "./bin/mocha.js",
        },
```

And looking at your macro example in rules_js I've created a `mocha_test` macro looking like this:
```starlark
# Inspired by https://github.com/aspect-build/rules_js/blob/main/examples/macro/mocha.bzl
load("@npm//:mocha/package_json.bzl", "bin")

def mocha_test(name, srcs, args = [], env = {}, **kwargs):
    bin.mocha_test(
        name = name,
        args = args,
        data = srcs,
        env = env,
        **kwargs
    )
```

Since the tests are also written in typescript we create a ts_project where `srcs` contain the test files and then depend on the needed modules in `deps`. This works perfectly fine with rules_nodejs 5.7.0 so I'm wondering if I'm missing something while migrating. I see that the output looks different compared to rules_nodejs so I'm guessing I might need some kind of adaption to how rules_js expects things.

In rules_nodejs I'm using [js_library](https://bazelbuild.github.io/rules_nodejs/Built-ins.html#js_library) to create virtual packages that map against the paths configured in tsconfig.json where I set `package_name` to match the expected import. I don't see a similar option for js_library in rules_js so I guess I might need npm_link_package or similar but I've yet figured out how to make it work.

Running with rules_js 1.6.5, rules_ts 1.0.0-rc5, typescript 4.8.4, mocha 10.1.0.

**Comment by thesayyn** [comment]
https://github.com/aspect-build/rules_js/blob/main/examples/npm_package/packages/pkg_a/BUILD.bazel this might help you.

rules_nodejs and rules_js has different ways of linking 1p packages.  
https://github.com/aspect-build/bazel-examples/tree/main/pnpm-workspaces is an example demonstrating this.

**Comment by thesayyn** [comment]
@fredrik-w does this help you?

**Comment by fredrik-w** [comment]
So I finally had some time over to play around with this again. And I think I finally cracked it.
I was trying to the define the `npm_link_package` in the BUILD-file of the package itself, rather in the root of the project.

Thanks to your links and the comments in those BUILD-files I realized I was doing it wrong.

Linking it from the root of the package and then explicitly set visibility to `//:__subpackages__ ` suddenly made things start moving. However, it appears I need to depend on the name of the link rather than the module, which I can buy as it has the `package_name` from `npm_package` and thus matches the path-alias from `tsconfig.json`.
